flink-end-to-end-tests/test-scripts/common_yarn_docker.sh

```diff
@@ -29,6 +29,8 @@ MAX_RETRY_SECONDS=120
 CLUSTER_SETUP_RETRIES=3
 IMAGE_BUILD_RETRIES=5
 
+SUCCEEDED=1
```

Comment:
Instead of introducing this custom variable, would it make sense to let `_on_exit_callback` in `common_utils.sh` export the `$?` so that the on exit actions have access to it? That way the mechanism would be simpler since the user of this script does not have to remember to set `SUCCEEDED` to 0 if everything was ok.

Reply:
```
function _on_exit_callback {
 export EXIT_CODE=$?
 ...
 // do the clean-up commands
}
```
If you mean the above the approach, i will verify whether it could work correctly both on linux and MAC. If yes, i will go in this way.

Reply:
I have verified that the exit code could be correctly trapped in the `_on_exit_callback`, both in linux like system or MAC. Please have another look at your convenience.